### PR TITLE
Dungeon Tracker improvements

### PR DESCRIPTION
**Pull request by frankdj100**
Improvements of 23-27 January, plus moving of all code to Dungeons.lua

### CODE REVIEW

**Comment by zdeyn** [comment]
I know it's a lot of code to be moved, but I think you got this

Noting that we share a problem with a lack of internationalization for the mob names, and we'll both need to address that in the future

File: Dungeons.lua [hunk 1]
```diff
@@ -0,0 +1,842 @@
...
+	if (Hardcore_Character.dt.sent_pulse ~= nil) and 
+	   (now - Hardcore_Character.dt.sent_pulse < DT_GROUP_PULSE) then
+		return
+	end
+	Hardcore_Character.dt.sent_pulse = now
+
+	-- Send my own info to the party (=name + server time + dungeon)
+	if( CTL ) then
+		local name, serverName = UnitFullName("player")
+		local commMessage = COMM_COMMANDS[15] .. COMM_COMMAND_DELIM .. name .. COMM_FIELD_DELIM .. now .. COMM_FIELD_DELIM .. Hardcore_Character.dt.current.name
```
Comment (aaronma37) [inline Dungeons.lua:559]: Is this working after the refactor? `COMM_COMMANDS` is local to `hardcore.lua`

File: Dungeons.lua [hunk 2]
```diff
@@ -0,0 +1,842 @@
...
+
+-- Function to do the unitscan trick of seeing nearby NPC for dungeon tracking
+-- Called from the ADDON_ACTION_FORBIDDEN event handler in Hardcore.lua
+function DungeonTrackerHandleActionForbidden(arg1)
+	if arg1 == 'Hardcore' then
+		dt_forbidden = true
+	end
+end
+
+function DungeonTrackerCheckChanged( name )
```
Comment (aaronma37) [inline Dungeons.lua:414]: looks like this can be local

File: Dungeons.lua [hunk 3]
```diff
@@ -0,0 +1,842 @@
...
+			end			
+		end
+	end
+end
+
+-- DungeonTrackerSendPulse( now )
+--
+-- Sends a group pulse, if the time out is expired
+
+function DungeonTrackerSendPulse( now )
```
Comment (aaronma37) [inline Dungeons.lua:547]: local

File: Dungeons.lua [hunk 4]
```diff
@@ -0,0 +1,842 @@
...
+		CTL:SendAddonMessage("NORMAL", COMM_NAME, commMessage, "PARTY")			-- Maybe to "INSTANCE_CHAT" instead?
+	end
+
+end
+
+-- DungeonTracker
+--
+-- Main interface function for the dungeon tracker, called on a 1s second timer
+
+function DungeonTracker()
```
Comment (aaronma37) [inline Dungeons.lua:569]: local

File: Dungeons.lua [hunk 5]
```diff
@@ -0,0 +1,842 @@
...
+			end
+			table.insert( the_table, { v[3], max_era_level, v[7][2] } )
+		end
+	end
+	
+	return the_table
+end
+
+
+function DungeonTrackerPopulateFromQuests()
```
Comment (aaronma37) [inline Dungeons.lua:208]: local

File: Dungeons.lua (hunk 1 above)
Comment (frankdj100) [reply Dungeons.lua:559]: Weird, I am sure it got called, but didn't throw a script error. But fixed, anyway.

Comment (frankdj100) [reply Dungeons.lua:559]: Wait, CTL doesn't exist in Dungeons.lua, so the if statement failed

Comment (frankdj100) [reply Dungeons.lua:559]: Alright, needed to copy over a bunch of COMM_ defines from Hardcore.lua (that weren't exported). Works now.

**Pull request merged**